core/Updater/Migration/Db/ChangeColumnTypes.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Updater\Migration\Db;
+
+/**
+ * @see Factory::changeColumnTypes()
+ * @ignore
+ */
+class ChangeColumnTypes extends Sql
+{
+    public function __construct($table, $columns)
+    {
+        $changes = array();
+        foreach ($columns as $columnName => $columnType) {
+            $changes[] = sprintf("CHANGE `%s` `%s` %s", $columnName, $columnName, $columnType);
+        }
+
+        $sql = sprintf("ALTER TABLE `%s` %s", $table, implode(', ', $changes));
+
+        parent::__construct($sql, static::ERROR_CODE_DUPLICATE_COLUMN);
```

Comment:
The duplicate error maps to 1060 and this is what we used to use see eg https://github.com/piwik/piwik/pull/10028/files/756655316f1e2cfc9678e7c110ca9ce003f2c9c9#diff-b3898c538dda4ef48012ac8ac8ba4c16L28 . It sounds valid to ignore the error you mentioned but not sure why it was never done